src/tool_cb_hdr.c

```diff
@@ -94,39 +95,50 @@ size_t tool_header_cb(char *ptr, size_t size, size_t nmemb, void *userdata)
      url && (checkprefix("http://", url) || checkprefix("https://", url))) {
     const char *p = str + 20;
 
-    /* look for the 'filename=' parameter
-       (encoded filenames (*=) are not supported) */
+    /* look for the 'filename=' or 'filename*=' parameters
+       encoded filenames (*=) are prioritized */
     for(;;) {
       char *filename;
       size_t len;
+      bool encoded = false;
 
       while(*p && (p < end) && !ISALPHA(*p))
         p++;
       if(p > end - 9)
         break;
 
-      if(memcmp(p, "filename=", 9)) {
+      if(memcmp(p, "filename", 8)) {
         /* no match, find next parameter */
         while((p < end) && (*p != ';'))
           p++;
         continue;
       }
-      p += 9;
+      p += 8;
```

Comment:
this needs to be strict, you removed the check for the = so filenamefoo for example would be incorrectly accepted